en_US/security/authn/authn.md

```diff
@@ -1,243 +1,812 @@
+| password_based   | postgresql         | [Authenticaton with PostgreSQL database as credential storage](postgresql.md) |
+| password_based   | mongodb            | [Authenticaton with MongoDB database as credential storage](./mongodb.md)     |
+| password_based   | redis              | [Authenticaton with Redis database as credential storage](./redis.md)         |
+| password_based   | http               | [Authenticaton using external HTTP API for credential verification](./http.md)|
+| jwt              |                    | [Authenticaton using JWT](./jwt.md)                                           |
+| scram            | built_in_database  | [Authenticaton using SCRAM](./scram.md)                                       |
+
+Each authenticator has its own configuration options.
+
+Example:
+
+```hocon
+{
+  mechanism = password_based
+  password_hash_algorithm {
+    name = plain
+    salt_position = suffix
+  }
+  enable = true
+  backend = redis
+  cmd = "HMGET mqtt_user:${username} password_hash salt is_superuser"
+  database = 1
+  password = public
+  redis_type = single
+  server = "127.0.0.1:6379"
+}
+```
 
-**Built-in data source**
+When a client connects, the configured authenticators perform identity verification of the client by checking
+whether its `username`/`clientid` and `password` are consistent with the data pertaining to the authenticator.
+
+## Authentication chains
+
+When authenticating a client, EMQX may try to perform identity verification using several authenticators
+sequentially. Each authenticator may either return authentication success/failure or pass verification to
+the next authenticator in the sequence. Such sequences are called _authentication chains_.
+
+Conditions on which verification is passed further in the chain depend on the authenticator and described
+in its documentation.
+
+Each authentication chain can contain only one authenticator of each type.
+
+Configured authentication chains can be updated or rearranged dynamically. Each authenticator
+can be _enabled_ or _disabled_.
+
+Example:
+
+```hocon
+authentication = [
+  {
+    mechanism = password_based
+    password_hash_algorithm {
+      name = plain
+      salt_position = suffix
+    }
+    enable = true
+    backend = redis
+    cmd = "HMGET mqtt_user:${username} password_hash salt is_superuser"
+    database = 1
+    password = public
+    redis_type = single
+    server = "127.0.0.1:6379"
+  },
+  {
+    mechanism = password_based
+    backend = built_in_database
+    user_id_type = username
+    password_hash_algorithm {
+        name = sha256
+        salt_position = suffix
+    }
+    enable = false
+  }
+]
 
+```
 
-{% emqxee %}
+An authentication chain without any enabled authenticators allows anonymous access.
 
-* [Built-in Auth](../modules/mnesia_authentication.md)
+Authentication chains can be configured globally per protocol as well as per individual protocol listeners:
```

Comment:
Nit: maybe say whether the specific chains compose with or completely override the global chain.